plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/marker/resolution/ChangeDataTypeMarkerResolution.java

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Sebastian Hollersbacher
+ *    - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.application.marker.resolution;
+
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.fordiac.ide.model.commands.change.ChangeDataTypeCommand;
+import org.eclipse.fordiac.ide.model.commands.change.ChangeStructCommand;
+import org.eclipse.fordiac.ide.model.commands.change.ConfigureFBCommand;
+import org.eclipse.fordiac.ide.model.data.DataType;
+import org.eclipse.fordiac.ide.model.errormarker.FordiacErrorMarker;
+import org.eclipse.fordiac.ide.model.libraryElement.ConfigurableFB;
+import org.eclipse.fordiac.ide.model.libraryElement.IInterfaceElement;
+import org.eclipse.fordiac.ide.model.libraryElement.StructManipulator;
+import org.eclipse.fordiac.ide.model.search.AbstractLiveSearchContext;
+import org.eclipse.fordiac.ide.model.typelibrary.ErrorDataTypeEntry;
+import org.eclipse.fordiac.ide.model.ui.editors.DataTypeTreeSelectionDialog;
+import org.eclipse.fordiac.ide.model.ui.nat.DataTypeSelectionTreeContentProvider;
+import org.eclipse.fordiac.ide.model.ui.nat.TypeNode;
+import org.eclipse.fordiac.ide.ui.FordiacMessages;
+import org.eclipse.jface.window.Window;
+import org.eclipse.ui.PlatformUI;
+
+public class ChangeDataTypeMarkerResolution extends AbstractErrorMarkerResolution {
+
+	private DataType selectedType;
+	private boolean canceled = false;
+
+	public ChangeDataTypeMarkerResolution(final IMarker marker) {
+		super(FordiacMessages.Repair_Dialog_ChangeDataType, marker);
+	}
+
+	@Override
+	public void run(final IMarker marker) {
+		if (canceled) {
+			return;
+		}
+
+		if (selectedType == null) {
+			selectDataType();
+		}
+
+		if (selectedType != null) {
+			final EObject errorType = FordiacErrorMarker.getTarget(marker);
+			if (errorType instanceof final IInterfaceElement interfaceElement
+					&& interfaceElement.getType().getTypeEntry() instanceof ErrorDataTypeEntry) {
```

Comment:
This may cause an NPE in case the reason for the error is that the type is null. Also, why do you need to explicitly check if it is an `ErrorDataTypeEntry` here in the first place?